lib/ui/zoomable_live_image.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:video_player/video_player.dart';
+
+class ZoomableLiveImage extends StatefulWidget {
+  final File photo;
+  final Function(bool) shouldDisableScroll;
+  final String tagPrefix;
+  final Decoration backgroundDecoration;
+
+  ZoomableLiveImage(
+    this.photo, {
+    Key key,
+    this.shouldDisableScroll,
+    @required this.tagPrefix,
+    this.backgroundDecoration,
+  }) : super(key: key);
+
+  @override
+  _ZoomableLiveImageState createState() => _ZoomableLiveImageState();
+}
+
+class _ZoomableLiveImageState extends State<ZoomableLiveImage>
+    with SingleTickerProviderStateMixin {
+  final Logger _logger = Logger("ZoomableLiveImage");
+  File _livePhoto;
+  bool _loadLivePhotoVideo = false;
+
+  VideoPlayerController _videoPlayerController;
+  ChewieController _chewieController;
+
+  @override
+  void initState() {
+    _livePhoto = widget.photo;
+    _loadLiveVideo();
+    super.initState();
+  }
+
+  void _onLongPressEvent(bool isPressed) {
+    if (_videoPlayerController != null && isPressed == false) {
+      // stop playing video
+      _videoPlayerController.pause();
+    }
+    if (mounted) {
+      setState(() {
+        _loadLivePhotoVideo = isPressed;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget content;
+    if (_loadLivePhotoVideo && _videoPlayerController != null) {
+      content = _getVideoPlayer();
+    } else {
+      content = ZoomableImage(_livePhoto,
+          tagPrefix: widget.tagPrefix,
+          shouldDisableScroll: widget.shouldDisableScroll,
+          backgroundDecoration: widget.backgroundDecoration);
+    }
+    return GestureDetector(
+        onLongPressStart: (_) => {_onLongPressEvent(true)},
+        onLongPressEnd: (_) => {_onLongPressEvent(false)},
+        child: content);
+  }
+
+  @override
+  void dispose() {
+    if (_videoPlayerController != null) {
+      _videoPlayerController.pause();
+      _videoPlayerController.dispose();
+    }
+    if (_chewieController != null) {
+      _chewieController.dispose();
+    }
+    super.dispose();
+  }
+
+  Widget _getVideoPlayer() {
+    _videoPlayerController.seekTo(Duration.zero);
+    _chewieController = ChewieController(
+        videoPlayerController: _videoPlayerController,
+        aspectRatio: _videoPlayerController.value.aspectRatio,
+        autoPlay: true,
+        autoInitialize: true,
+        looping: false,
+        allowFullScreen: false,
+        showControls: false);
+    return Chewie(controller: _chewieController);
+  }
+
+  void _loadLiveVideo() {
+    if (_videoPlayerController != null) {
+      return;
+    }
+    getFile(widget.photo, liveVideo: true).then((file) {
+      if (file != null && file.existsSync()) {
+        _logger.fine("loading  from local");
+        _setVideoPlayerController(file: file);
+      } else if (widget.photo.uploadedFileID != null) {
+        _logger.fine("loading from remote");
+        getFileFromServer(widget.photo, liveVideo: true).then((file) {
```

Comment:
Doesn't `getFile` internally call `getFileFromServer`?

Reply:
Okay I guess if it thinks it's locally available and then fails, we'll have to reload from server. We should think about placing this logic within `getFile` itself, so that the view layer can be agnostic to this. `todo` for future. 🤷